generator/src/main/java/com/graphicsfuzz/generator/tool/PrepareReference.java

```diff
@@ -135,8 +137,10 @@ public static void prepareReference(ShaderJob shaderJob,
           shaderJob.getPipelineInfo());
     }
 
-    if (maxUniforms > 0 && shaderJob.getPipelineInfo().getNumUniforms() > maxUniforms) {
-      throw new RuntimeException("Too many uniforms in reference shader job.");
+    if (maxUniforms > 0) {
+      // The reference shader job has more uniforms than the specified limit, so prune some of them
+      // by inlining their values into the shaders.
+      PruneUniforms.prune(shaderJob, maxUniforms, Collections.emptyList());
```

Comment:
This comment is a bit confusing, as it is not necessarily true. And perhaps `prune` should be `pruneIfNeeded`?

Reply:
Agreed on both counts - done.